test/e2e/knative_serving_test.go

```diff
@@ -191,7 +188,12 @@ func testKnativeVersusKubeServicesInOneNamespace(t *testing.T, caCtx *test.Conte
 	caCtx.Clients.Serving.ServingV1beta1().Services(testNamespace2).Delete(ksvc.Name, &metav1.DeleteOptions{})
 }
 
-func testUserPermissions(t *testing.T, paCtx *test.Context, editCtx *test.Context, viewCtx *test.Context) {
+func testUserPermissions(t *testing.T) {
+	paCtx := test.SetupProjectAdmin(t)
+	editCtx := test.SetupEdit(t)
+	viewCtx := test.SetupView(t)
+	test.CleanupOnInterrupt(t, func() { test.CleanupAll(paCtx, editCtx, viewCtx) })
```

Comment:
I think this needs a call to CleanupAll on `defer` as well to properly clean up.

Reply:
Umm... If we call it by defer, we will not be able to get the logs.

Is it not better to expect to clean up them when CR is deleted?

Reply:
In 0.13 we won't clean everything when the CR gets removed anymore. Which logs do we not get in this case?

Reply:
Sorry, not log but the status of Ksvc created by editCtx, editCtx or viewCtx. But I change my mind and I think it is not necessary to collect. I will add defer.